warp/Network/Wai/Handler/Warp/FdCache.hs

```diff
@@ -26,7 +26,7 @@ import System.Posix.IO (
     openFd,
     setFdOption,
  )
-import UnliftIO.Exception (bracket)
+import Control.Exception (bracket)
```

Comment:
The `UnliftIO` bracket is different in that it uses `uninterruptibleMask_` on the `after` actions. How important are the `terminate` actions?
They are cleaning up file descriptors, so that might be important enough to absolutely make sure it happens, right?

Reply:
Resouce-release actions in `bracket` (i.e. `mask`ed actions) are iterruptible only if they are blocked (such as `takeMVar`).
Non-blocking actions are NOT iterruptible.
In other words, asynchronous exceptions are not delivered to resource-release actions if they are non-blocking.

`terminate` is non-blocking, so it's free from asynchronous exceptions, I believe.

Reply:
Ah, yes. Never mind. I've re-read the `bracket` docs, and it does. I misread, I think.